jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/client/HttpClientProvider.java

```diff
@@ -16,23 +16,43 @@
 //  ========================================================================
 //
 
-package org.eclipse.jetty.websocket.client.impl;
+package org.eclipse.jetty.websocket.core.client;
 
 import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.util.log.Log;
 import org.eclipse.jetty.util.ssl.SslContextFactory;
 import org.eclipse.jetty.util.thread.QueuedThreadPool;
 
-class DefaultHttpClientProvider
+public interface HttpClientProvider
 {
-    public static HttpClient newHttpClient()
+    static HttpClient get()
     {
-        SslContextFactory sslContextFactory = new SslContextFactory();
-        HttpClient client = new HttpClient(sslContextFactory);
+        try
+        {
+            HttpClientProvider xmlProvider = new XmlHttpClientProvider();
+            HttpClient client = xmlProvider.newHttpClient();
+            if (client != null)
+                return client;
+        }
+        catch (Throwable ignore)
+        {
+            Log.getLogger(HttpClientProvider.class).ignore(ignore);
+        }
+
+        return HttpClientProvider.newDefaultHttpClient();
+    }
+
+    private static HttpClient newDefaultHttpClient()
+    {
+        HttpClient client = new HttpClient(new SslContextFactory());
```

Comment:
Note that we were previously setting setEndpointIdentificationAlgorithm("HTTPS").    Should we still do that or has it been changed to the default now?

Reply:
Yes the `endpointIdentificationAlgorithm` now defaults to "HTTPS" in the `SslContextFactory`.
I had this line in previously but @sbordet said it was not needed.